core/network/impl/gossiper_server_libp2p.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "network/impl/gossiper_server_libp2p.hpp"
+
+#include "libp2p/basic/message_read_writer.hpp"
+#include "libp2p/connection/stream.hpp"
+#include "network/impl/common.hpp"
+#include "scale/scale.hpp"
+
+namespace kagome::network {
+  using libp2p::basic::MessageReadWriter;
+
+  GossiperServerLibp2p::GossiperServerLibp2p(libp2p::Host &host,
+                                             common::Logger log)
+      : host_{host}, log_{std::move(log)} {}
+
+  void GossiperServerLibp2p::start() {
+    host_.setProtocolHandler(
+        kGossipProtocol, [self{shared_from_this()}](auto stream) {
+          self->handleGossipProtocol(
+              std::make_shared<MessageReadWriter>(stream), std::move(stream));
```

Comment:
I don't like how gossiper server is organized. 
You might implement a session, which could keep read_writer instance instead of passing it to each function.
Code would be more clear and server code would be neat, e.g. it took me some time to realize why read_writer stays alive.
Of course you would have to pass logger and block announce handler to session constructor.